test/server/cards/03_TWI/units/GeneralsGuardian.spec.ts

```diff
@@ -22,9 +22,19 @@ describe('General\'s Guardian', function() {
 
                 const battleDroids = context.player2.findCardsByName('battle-droid');
                 expect(battleDroids.length).toBe(1);
-                expect(battleDroids).toAllBeInZone('groundArena');
+                expect(battleDroids).toAllBeInZone('groundArena', context.player2);
                 expect(battleDroids.every((battleDroid) => battleDroid.exhausted)).toBeTrue();
             });
+
+            it('should not create a Battle Droid when a different unit is attacked', function () {
```

Comment:
Can merge this into the above test pretty easily, same for Knight of the Republic

Reply:
fixed up